azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -494,20 +494,24 @@ VersionSet fetchVersionSet(final int executionId, final Map<String, String> flow
             if (flowParams.containsKey(FlowParameters.FLOW_PARAM_ALLOW_IMAGE_TEST_VERSION) &&
                 Boolean.TRUE.equals(Boolean
                     .valueOf(flowParams.get(FlowParameters.FLOW_PARAM_ALLOW_IMAGE_TEST_VERSION)))) {
-              versionMap.put(imageType,
+              overlayMap.put(imageType,
                   this.imageRampupManager.getVersionInfo(imageType,
                       flowParams.get(imageTypeVersionOverrideParam),
                       State.getNewActiveAndTestStateFilter()));
             } else {
-              versionMap.put(imageType,
+              overlayMap.put(imageType,
                   this.imageRampupManager.getVersionInfo(imageType,
                       flowParams.get(imageTypeVersionOverrideParam),
                       State.getNewAndActiveStateFilter()));
             }
           }
         }
 
+        final Map<String, VersionInfo> versionMap =
+            this.imageRampupManager.getVersionByImageTypes(executableFlow, imageTypesUsedInFlow,
+                overlayMap.keySet());
         final VersionSetBuilder versionSetBuilder = new VersionSetBuilder(this.versionSetLoader);
+        versionSetBuilder.addElements(overlayMap);
```

Comment:
What is given priority? Let's say there is an active version and then new version is mentioned from flow param. As you are adding overlayMap first and then versionMap, will it give preference to ACTIVE version?

Reply:
You're right. Let me reverse the order.